### ISSUE TITLE
Retroarch "freezes" ingame and in menus

### ISSUE THREAD

**Issue by Deivmsr** [libretro/RetroArch#9737]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
Retroarch is "freezing" in running games and also when accessing menus

### Expected behavior

What I expected to happen is what I could notice until the d84dfee commit, ie letting you in and out of any game or option normally.

### Actual behavior
Retroarch completely freezes when going to "menu scale factor" menu, ie it does not allow you to press the back button of the "bar" or menu bar to go to the home page. 
Another problem is that when entering a game in any core Retroarch completely freezes completely, in which case it does not let you go back to the quick menu, ie it freezes well on the screen of the game that is still running. The only way out of Retroarch is by hitting the "home" button and closing the emulator through the Android "recent apps" menu.

### Steps to reproduce the bug

1. [First step]
2. [Second step]
3. [and so on...]

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
Version 1.8.1

- RetroArch: [version/commit]

Commit - 0d284c4 - nightly build

- OS: Android 9.0/Galaxy Note 9 (SD version)

**Comment by twinaphex** [comment]
That Github commit you point to really cannot be the cause in any way or shape.

It must be something else.

But honestly, I have a hard time making sense of this report.

@jdgleaver Can you reproduce any of this?

**Comment by jdgleaver** [comment]
I can't reproduce any freezing in-menu, but I guess it might be possible that the freezing in-game may have been due to the input polling PR. I know that Android can behave oddly if the input is inhibited in any way - e.g. on my phone, if Flycast has the wrong input polling method the device will become completely unresponsive and will periodically pop up a force close request.

@Deivmsr We recently reverted a commit that affected input - could you please try again with the latest nightly?

**Comment by Deivmsr** [comment]
jdgleaver The freeze issue on the virtual gamepad was fixed in the nightly build I tested. Now the only bug I'm having is "appearance", where after selecting another option in "menu scale factor" it only accepts to go back to the previous menu after ticking the factor if any option is active.

**Comment by jdgleaver** [comment]
@Deivmsr Oh wow! Okay, I've managed to reproduce the 'menu scale factor' issue now - it's an incredibly obscure edge case that causes the current selection index to go our of bounds. This is actually harmless, with the only consequence being that the 'back' command gets disabled (since there must be a valid selection in order for it work - if you manually select any item, the function is restored again).

I've fixed this, and it will be submitted as part of my next PR. Thanks for reporting the issue - I don't think I ever would have noticed it otherwise!
